pkg/parser/yamlparser.go

```diff
@@ -90,31 +91,31 @@ func (doc *YamlDocument) ValidateYAML() {
 	}
 }
 
-func GetParsedYAMLWithURI(URI protocol.URI) (YamlDocument, error) {
+func ParseURI(URI protocol.URI, context *utils.LsContext) (YamlDocument, error) {
```

Comment:
ParseURI has the same issue imo, parse uri means parsing an uri, not a file linked to an URI

Reply:
What about `ParseFromURI` ?